src/XMLimport.cpp

```diff
@@ -563,27 +563,27 @@ pair<int, int> XMLimport::readPackage()
             if (name() == "HostPackage") {
                 readHostPackage();
             } else if (name() == "TriggerPackage") {
-                objectType = dlgTriggerEditor::cmTriggerView;
+                objectType = int(dlgTriggerEditor::EditorViewType::cmTriggerView);
```

Comment:
As @vadi2 says I do not _think_ the case is necessary and you should not use a 'C' style case anyhow - if it __IS__ necessary use the C++ static_cast i.e.:

    objectType = static_cast<int>(dlgTriggerEditor::EditorViewType::cmTriggerView);

but I do not think you will need it - but I could be rong! 😜

Reply:
@SlySven I get errors when I don't convert it to int. The types don't match. Also, all the functions require int variables as arguments because of which I couldn't pass the enum variables directly.

Reply:
Ah, right you are; thinking about it I had the same issue when I tweaked the `TRoom::customLineStyles` from being a `QString` to a `Qt::PenStyle` in the last couple of days in my big GUI Internationalisation PR - so you will need the cast but use the C++ one anyhow.

For what it is worth, do you need all the namespaces in there BTW - with the full `QFlag` treatment from the declare flags macros you lose the need for the middle `EditorviewType` if I understand it correctly?

Reply:
I think the namespaces are the entire point of C++11 class enums - they're no longer polluting the global namespace and constraining us with having to all be unique.

Reply:
@vadi2 So, what should I do exactly? Should I keep it?

Reply:
Yeah, I think this is good and going forward we'll be able to use different enums in the `dlgTriggerEditor` class without it being confusing. Let's wait to see what @SlySven says.

Reply:
That said what about the C++ `static_cast<int>`, that seems like a good idea.

Reply:
@vadi2 :+1: I'll change `int()` to `static_cast<int>`.